exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/partition/PruneScanRule.java

```diff
@@ -176,81 +177,103 @@ protected void doOnMatch(RelOptRuleCall call, DrillFilterRel filterRel, DrillPro
+          MaterializedField field = MaterializedField.create(column, type);
+          ValueVector v = TypeHelper.getNewVector(field, allocator);
+          v.allocateNew();
+          vectors[partitionColumnIndex] = v;
+          container.add(v);
+        }
 
-      if (errors.getErrorCount() != 0) {
-        logger.warn("Failure while materializing expression [{}].  Errors: {}", expr, errors);
-      }
+        // populate partition vectors.
+        descriptor.populatePartitionVectors(vectors, partitions, partitionColumnBitSet, fieldNameMap);
+
+        // materialize the expression
+        logger.debug("Attempting to prune {}", pruneCondition);
+        final LogicalExpression expr = DrillOptiq.toDrill(new DrillParseContext(settings), scanRel, pruneCondition);
+        final ErrorCollectorImpl errors = new ErrorCollectorImpl();
+
+        LogicalExpression materializedExpr = ExpressionTreeMaterializer.materialize(expr, container, errors, optimizerContext.getFunctionRegistry());
+        // Make sure pruneCondition's materialized expression is always of BitType, so that
+        // it's same as the type of output vector.
+        if (materializedExpr.getMajorType().getMode() == TypeProtos.DataMode.REQUIRED) {
+          materializedExpr = ExpressionTreeMaterializer.convertToNullableType(
+              materializedExpr,
+              materializedExpr.getMajorType().getMinorType(),
+              optimizerContext.getFunctionRegistry(),
+              errors);
+        }
 
-      output.allocateNew(partitions.size());
-      InterpreterEvaluator.evaluate(partitions.size(), optimizerContext, container, output, materializedExpr);
-      int record = 0;
+        if (errors.getErrorCount() != 0) {
+          logger.warn("Failure while materializing expression [{}].  Errors: {}", expr, errors);
+        }
 
-      List<String> newFiles = Lists.newArrayList();
-      for(PartitionLocation part: partitions){
-        if(!output.getAccessor().isNull(record) && output.getAccessor().get(record) == 1){
-          newFiles.add(part.getEntirePartitionLocation());
+        output.allocateNew(partitions.size());
+        InterpreterEvaluator.evaluate(partitions.size(), optimizerContext, container, output, materializedExpr);
+        int recordCount = 0;
+        int qualifiedCount = 0;
+
+        // Inner loop: within each batch iterate over the PartitionLocations
+        for(PartitionLocation part: partitions){
+          if(!output.getAccessor().isNull(recordCount) && output.getAccessor().get(recordCount) == 1){
+            newFiles.add(part.getEntirePartitionLocation());
+            qualifiedCount++;
+          }
+          recordCount++;
+        }
+        logger.debug("Within batch {}: total records: {}, qualified records: {}", batchIndex, recordCount, qualifiedCount);
+        batchIndex++;
+      } catch (Exception e) {
+        logger.warn("Exception while trying to prune partition.", e);
```

Comment:
If there is Exception during partition pruning for one sublist, seems we just log the error as a warning. That means, the code could continue the logic after the "for-loop". I feel that might produce incorrect result, since the list of new partitions might be invalid. 

Should we stop the execution of partition pruning, once an Exception is caught here?